Gems/Atom/Tools/ShaderManagementConsole/Code/Source/Window/ShaderManagementConsoleWindow.cpp

```diff
@@ -37,6 +38,52 @@ namespace ShaderManagementConsole
         Base::OnDocumentOpened(documentId);
         m_documentInspector->SetDocumentId(documentId);
     }
+
+    void ShaderManagementConsoleWindow::OnDocumentModified(const AZ::Uuid& documentId)
+    {
+        Base::OnDocumentModified(documentId);
+        m_documentInspector->SetDocumentId(documentId);
+    }
+
+    AZStd::string ShaderManagementConsoleWindow::GetSaveDocumentParams(const AZStd::string& initialPath) const
+    {
+        // Get shader file path
+        AZStd::string shaderFullPath;
+        AZ::RPI::ShaderVariantListSourceData shaderVariantList = {};
+        ShaderManagementConsoleDocumentRequestBus::EventResult(
+            shaderVariantList,
+            m_documentInspector->GetDocumentId(),
+            &ShaderManagementConsoleDocumentRequestBus::Events::GetShaderVariantListSourceData);
+        shaderFullPath = AZ::RPI::AssetUtils::ResolvePathReference(initialPath, shaderVariantList.m_shaderFilePath);
+        
+        QMessageBox msgBox;
+        msgBox.setText("Where do you want to save the list?");
+        QPushButton* projectBtn = msgBox.addButton(QObject::tr("Save to project"), QMessageBox::YesRole);
+        QPushButton* engineBtn = msgBox.addButton(QObject::tr("Save to engine"), QMessageBox::NoRole);
+        msgBox.exec();
+
+        AZStd::string result;
+        if (msgBox.clickedButton() == projectBtn)
+        {
+            char projectRoot[AZ_MAX_PATH_LEN];
+            AZ::IO::FileIOBase::GetInstance()->ResolvePath("@projectroot@", projectRoot, AZ_MAX_PATH_LEN);
+
+            AZStd::string shaderName;
+            AZStd::string shaderPath;
+            AzFramework::StringFunc::Path::Split(shaderFullPath.c_str(), nullptr, &shaderPath, &shaderName);
+            AZStd::string relativePath = shaderPath.substr(shaderPath.find("Assets"), shaderPath.length());
+            AzFramework::StringFunc::Path::StripComponent(relativePath);
+            result = AZStd::string::format("%s\\ShaderVariants\\%s\\%s.shadervariantlist", projectRoot, relativePath.c_str(), shaderName.c_str());
+            AzFramework::StringFunc::Path::Normalize(result);
+        }
```

Comment:
There are some utility functions like AZ::Utils::GetProjectPath() that might be helpful to simplify things

Reply:
I will use AZ::Utils::GetProjectPath() instead, though I didn't find a good way to get the shader relative path.